Classes/Sizes/Rootline.php

```diff
@@ -38,145 +36,111 @@ final class Rootline
 
     private BackendLayout $backendLayout;
 
-    private array $rootline = [];
+    private ContainerRepository $containerRepository;
 
     private array $finalSizes = [];
 
     private string $fieldName;
 
-    public function __construct(array $data, string $fieldName)
-    {
+    private string $backendLayoutIdentifier;
+
+    public function __construct(
+        array $data,
+        string $fieldName
+    ) {
+        $this->containerRepository = new ContainerRepository();
+
         $this->determineBackendLayout();
         $this->fieldName = $fieldName;
-        $this->contentElement = $this->determineContentElement($data);
+        $this->contentElement = $this->determineContentElement(null, $data);
+
+        $this->determineRootline($this->contentElement);
 
-        $this->determineRootline();
-        $this->calculateSizes();
+        $this->finalSizes = $this->contentElement->getFinalSize([]);
     }
 
     public function getFinalSizes(): array
     {
-        return $this->finalSizes;
+        $sizes = $this->finalSizes;
+
+        foreach ($sizes as $sizeName => &$size) {
+            $size = ceil($size);
+        }
+
+        return $sizes;
+    }
+
+    public function getSizesAndMultiplierFromContentElement(
```

Comment:
I couldn't find any usage of this method. Do we still need the method?

Reply:
Sollte erledigt sein.